### ISSUE TITLE
Ability to Search for and Group Tilesets

### ISSUE THREAD

**Issue by benjamin-kirkbride** [mapeditor/tiled#3577]
**Is your feature request related to a problem? Please describe.**
I have a ton of tilesets (literally hundreds, don't ask) and would like to be able to search through them and also categorize them.

**Describe the solution you'd like**
Any combination of:
- a search
- a tagging system
- folders
- ordering

**Describe alternatives you've considered**
excruciating scrolling

**Comment by eishiya** [comment]
I thought there was an issue for this already, but apparently not!
Related feature requests: #873, #2901

For now, there is the Projects panel and the Search In Project (default shortcut: Ctrl+P), which can be used bring up the relevant tilesets by their location or name. However, while this is useful for opening a tileset, it won't help if the problem is that you already have too many tilesets open to begin with.

I think a related feature request would be the option to hide Tilesets panel tabs for tilesets that are not part of the current map.

**Comment by benjamin-kirkbride** [comment]
Yeah certainly a lot of discussion to be had about exactly how this should/could be implemented.

Is anyone familiar with the JS or Python Plugins able to speak to whether this could be added via that?

**Comment by benjamin-kirkbride** [comment]
Looking through the docs, it seems that it's only possible to create temporary dialogs/modals, and that it is not possible to create new views.

Is this correct?

https://www.mapeditor.org/docs/scripting/

**Comment by eishiya** [comment]
That is correct.

It's possible to make a Dialog that lets you search currently open tilesets based on custom properties and bring up a single tileset as the currently active tileset in the Tilesets panel though, which should cover most needs. If multiple matches are found, the dialog can display a list for you to select one from.

**Comment by benjamin-kirkbride** [comment]
I see, but you can't put it into the UI like the Layers Panel, for instance. Perhaps a new issue should be made for that haha
